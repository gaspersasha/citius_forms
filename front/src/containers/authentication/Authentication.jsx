import React from 'react';
import { AuthWrapper, AuthLink } from '.';

import s from './auth.module.sass';

const Authentication = () => (
  <AuthWrapper>
    {({ openModal }) => (
      <ul className={s.authButtonsWrapper}>
        <AuthLink dataId="button-login" openModal={() => openModal('login')}>
          Login
        </AuthLink>
        <AuthLink dataId="button-signup" openModal={() => openModal('signup')}>
          Sign Up
        </AuthLink>
      </ul>
    )}
  </AuthWrapper>
);

export default Authentication;
