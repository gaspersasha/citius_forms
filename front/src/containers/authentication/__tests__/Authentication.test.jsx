import React from 'react';
import Authentication from '..';

global.mockUseUserContext({
  actions: { logout: jest.fn() },
});

describe('<Authentication />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Authentication />);
  });

  it('should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have button for open AuthModal', () => {
    ['login', 'signup'].forEach((selector) => {
      const elem = wrapper
        .find(`span[data-id="button-${selector}"]`)
        .simulate('click');
      const overlay = wrapper.find('div[data-id="overlay"]');

      expect(overlay.length).toBe(1);
      expect(elem).toHaveLength(1);
    });
  });
});
