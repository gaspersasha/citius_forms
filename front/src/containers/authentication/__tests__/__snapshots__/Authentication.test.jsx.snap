// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication /> should match snapshots 1`] = `
<Authentication>
  <AuthWrapper>
    <ul
      className="authButtonsWrapper"
    >
      <AuthenticationLink
        className=""
        dataId="button-login"
        openModal={[Function]}
      >
        <li
          className="leaf"
        >
          <span
            className="link"
            data-id="button-login"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Login
          </span>
        </li>
      </AuthenticationLink>
      <AuthenticationLink
        className=""
        dataId="button-signup"
        openModal={[Function]}
      >
        <li
          className="leaf"
        >
          <span
            className="link"
            data-id="button-signup"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            Sign Up
          </span>
        </li>
      </AuthenticationLink>
    </ul>
  </AuthWrapper>
</Authentication>
`;
